certora/specs/sharesAssetsConversion.spec

```diff
@@ -0,0 +1,257 @@
+	assert convertToAssets(shares) <= amount, "Too many converted shares";
+	assert convertToAssets(shares + 1) >= amount, "Too few converted shares";
+}
+
+
+// Converting shares to amount is rounded down.
+rule sharesConversionRoundedDown(uint256 shares) {
+	uint256 amount = convertToAssets(shares);
+	assert convertToShares(amount) <= shares, "Amount converted is too high";
+	assert convertToShares(amount + 1) >= shares, "Amount converted is too low";
+}
+
+
+// Converting amount to shares and back to amount is preserved (up to rounding).
+/* NOTE: The precision depends on the ratio rate/ray.
+ */
+rule amountConversionPreserved(uint256 amount) {
+ 	require rate() <= RAY();
+ 	mathint mathamount = to_mathint(amount);
+ 	mathint converted = to_mathint(convertToAssets(convertToShares(amount)));
+
+	// That converted <= mathamount was proved in amountConversionRoundedDown,
+ 	assert mathamount - converted <= 1 + rate() / RAY(), "Too few converted assets";
+ }
+ 
+
+// Converting shares to amount and back to shares is preserved up to RAY.
+/* NOTE: The precision depends on the ratio RAY/rate.
+ */
+rule sharesConversionPreserved(uint256 shares) {
+	mathint mathshares = to_mathint(shares);
+	uint256 amount = convertToAssets(shares);
+	mathint converted = to_mathint(convertToShares(amount));
+
+	// That converted <= mathshare was proved in sharesConversionRoundedDown.
+	assert mathshares - converted <= 1 + RAY() / rate(), "Too few converted shares";
+}
+
+
+/* Joining and splitting accounts provides limited advantage.
+ * This rule verifies that joining accounts (by combining shares), and splitting accounts
+ * (by splitting shares between accounts) provides limited advantage when converting to
+ * asset amounts.
+ */
+rule accountsJoiningSplittingIsLimited(uint256 shares1, uint256 shares2) {
+    uint256 amount1 = convertToAssets(shares1);
+    uint256 amount2 = convertToAssets(shares2);
+    uint256 jointShares = shares1 + shares2;
+    require jointShares >= shares1 + shares2;  // Prevent overflow
+    uint256 jointAmount = convertToAssets(jointShares);
+
+    assert jointAmount >= amount1 + amount2, "Found advantage in combining accounts";
+    assert jointAmount < amount1 + amount2 + 2, "Found advantage in splitting accounts";
+    // The following assertion fails (as expected):
+    // assert jointAmount < amount1 + amount2 + 1, "Found advantage in splitting accounts";
+}
+
+// Similar rule as above for shares
+rule convertSumOfAssetsPreserved(uint256 assets1, uint256 assets2) {
+    uint256 shares1 = convertToShares(assets1);
+    uint256 shares2 = convertToShares(assets2);
+    uint256 sumAssets = assets1 + assets2;
+	require sumAssets >= assets1 + assets2; // Prevent overflow
+    uint256 jointShares = convertToShares(sumAssets);
+
+	assert jointShares >= shares1 + shares2, "Convert sum of assets bigger than parts";
+	assert jointShares < shares1 + shares2 + 2, "Convert sum of assets far smaller than parts";
```

Comment:
this is occurring due to rounding up of deposit right?
Im not sure this is ok to gain an extra share here, let's disucss it...

Reply:
Yes. I added a comment explaining this.